interpreter/cling/test/NullDeref/If.C

```diff
@@ -0,0 +1,24 @@
+//------------------------------------------------------------------------------
+// CLING - the C++ LLVM-based InterpreterG :)
+//
+// This file is dual-licensed: you can choose to license it under the University
+// of Illinois Open Source License or the GNU Lesser General Public License. See
+// LICENSE.TXT for details.
+//------------------------------------------------------------------------------
+
+// RUN: cat %s | %cling -Xclang -verify
+//This file checks an if statement for null prt dereference.
+
+#include <stdlib.h>
+
+int* p = (int*)0x1;
+
+if (*p) {} // expected-warning {{invalid memory pointer passed to a callee:}}
+
+if (true) { *p; exit(1); } // expected-warning {{invalid memory pointer passed to a callee:}}
+
+if (false) {
+} else { *p; exit(1); } // expected-warning {{invalid memory pointer passed to a callee:}}
```

Comment:
Can this go onto the previous line, just to make sure we know what happens? (And to signal it to the casual code reader.)